api/src/main/java/bio/terra/tanagra/utils/GoogleBigQuery.java

```diff
@@ -48,6 +49,20 @@ public GoogleBigQuery(GoogleCredentials credentials, String projectId) {
             .build()
             .getService();
     this.tableSchemasCache = new HashMap<>();
+    this.querySchemasCache = new HashMap<>();
+  }
+
+  public Schema getQuerySchemaWithCaching(String query) {
```

Comment:
this method is called in BigQueryDataset, which I assume (and I might be wrong) that it might be called from different threads but querySchemasCache is not thread safe.
If my assumption is correct, then querySchemasCache better be a ConcurrentHashMap

Reply:
Updated. It's not called from multiple threads now because the rudimentary CLI commands are always single-threaded. But in the future, I'd like to kickoff multiple indexing jobs in parallel to speed things up. You're right that I'd need a `ConcurrentHashMap` in that case, so this change will help for the future.